### PR TITLE
CosmosDB Create Account

### PR DESCRIPTION
**Pull request by olileger**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 101-cosmosdb-create-account/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      "metadata": {
+        "description": "CosmosDB account tier"
+      }
+    }
+  },
+  "resources": [
+    {
+      "name": "[parameters('name')]",
+      "type": "Microsoft.DocumentDB/databaseAccounts",
+      "apiVersion": "2015-04-08",
```
Comment (bmoore-msft) [inline 101-cosmosdb-create-account/azuredeploy.json]: update apiVersion

Comment (olileger) [reply 101-cosmosdb-create-account/azuredeploy.json]: Hello @bmoore-msft 
As per the ARM schema definition for CosmosDB there's currently only 1 available API version for this resource provider which is "2015-04-08"
[https://docs.microsoft.com/en-us/azure/templates/microsoft.documentdb/allversions](url)
[https://docs.microsoft.com/en-us/azure/templates/microsoft.documentdb/2015-04-08/databaseaccounts](url)

Am I wrong ?

Olivier

Comment (bmoore-msft) [reply 101-cosmosdb-create-account/azuredeploy.json]: the /providers api will tell you the available versions - you can use resource explorer in the portal or az providers list -o json in the cli to get the latest

**Comment by bmoore-msft** [comment]
There was a breaking change in a new version of node/npm - update your .travis.yml file to pick up that change or open a new PR after pulling from master.

File: 101-cosmosdb-create-account/azuredeploy.json (hunk 1 above)
Comment (olileger) [reply 101-cosmosdb-create-account/azuredeploy.json]: Ok by running the cmd line under Azure CLI I got the following output : 

`"apiProfiles": [
          {
            "apiVersion": "2015-04-08",
            "profileVersion": "2018-06-01-profile"
          }
        ],
        "apiVersions": [
          "2016-03-31",
          "2016-03-19",
          "2015-11-06",
          "2015-04-08",
          "2014-04-01"
        ],`

Should I use "2016-03-31" then ?
So question is about the apiVersions tab decorelated of the apiProfiles[0].apiVersion value ??
Other question is about how to well define the ARM template related to a version which is not documented at all on the Azure documentation (I may miss required field as well as not understanding the behavior of that template once deployed) ?
Regarding [https://docs.microsoft.com/en-us/azure/templates/microsoft.documentdb/2015-04-08/databaseaccounts](url) I don't have any other apiVersion from the combo box

Thanks

Comment (olileger) [reply 101-cosmosdb-create-account/azuredeploy.json]: > There was a breaking change in a new version of node/npm - update your .travis.yml file to pick up that change or open a new PR after pulling from master.

Could you clarify please ? Not sure to understand the link with my PR.
Thanks

Comment (olileger) [reply 101-cosmosdb-create-account/azuredeploy.json]: hello @bmoore-msft do you have any update please ? thanks
Olivier

Comment (bmoore-msft) [reply 101-cosmosdb-create-account/azuredeploy.json]: re: apiVersion - use the latest reported by the providers /api... docs can lag if the RP has not updated  recently.

re: the build break, look at the .travis.yml file in your branch - make sure it matches master.  If not you need to update it in your branch for CI to run.

Comment (olileger) [reply 101-cosmosdb-create-account/azuredeploy.json]: Hello @bmoore-msft 
It should be fixed now : apiVersion has been updated with the last available version.
Travis file has been updated with the master branch content.
Olivier

**Comment by bmoore-msft** [comment]
Can you check the CI failure (this one is not related to travis)

File: 101-cosmosdb-create-account/azuredeploy.json (hunk 1 above)
Comment (olileger) [reply 101-cosmosdb-create-account/azuredeploy.json]: @bmoore-msft 
It seems the `resourceGroup().location` is causing the issue but -correct me if I'm wrong- this is something recommended by you right here : [https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md](url)
Any suggestion please ?

Comment (olileger) [reply 101-cosmosdb-create-account/azuredeploy.json]: @bmoore-msft do you have any comment please ?
As defined per your guidelines : 

> Templates must have a parameter named location for the primary location of resources.
The default value of this parameter must be [resourceGroup().location]

But this is what is causing the issue right here.
BTW it was working without any error before the release pipeline update, at the time of the creation of the PR.

Thanks

Comment (olileger) [reply 101-cosmosdb-create-account/azuredeploy.json]: BTW are you trying to execute `az deployment validate` command ? this one fail but validating the template through `az group deployment validate` works perfectly

**Comment by olileger** [comment]
> Can you check the CI failure (this one is not related to travis)

The error "BadRequest : Invalid capability string" is not documented by Azure ARM template `https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-common-deployment-errors`

**Comment by olileger** [comment]
> Can you check the CI failure (this one is not related to travis)

Should be fixed now

**Comment by olileger** [comment]
Hello @bmoore-msft do you have any update please ?
Thanks

**Pull request merged**